GDPR Consent log

NicolasCador opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Make Log working for any module. <br> ...
| Type?         | bug fix / improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/Prestashop#31317
| How to test?  | Go to BO > Modules > GDPR and click on configure

Click on tab "Data Consent"

Enable GDPR on any module and Save

In FO, Check GDPR box and click on Submit

Go back to BO > Modules > GDPR > Data Consent

There is a log or registration or proof of your GDPR acceptance.



<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

fox-john commented:
Hi @NicolasCador,

The module has has been completely reworked on the PHP code side. You will probably need to rebase your work and check the compatibility of your code with the current code.

AlmostDigital commented:
Hello everyone, regarding the latest update I'm experiencing a db query error, the table prefix is written in text within the findAll() method and this generates an exception for table not found.


<img width="756" alt="Screenshot 2023-04-07 alle 13 30 38" src="https://user-images.githubusercontent.com/90357829/230602791-042fb3c2-5610-403e-9285-787305d8f69f.png">


I think it is necessary to include the constant _DB_PREFIX_,
best regards

NicolasCador committed:
Update FrontAjaxGdpr.php

NicolasCador committed:
Update gdpr.php

NicolasCador committed:
Update psgdpr.php

NicolasCador committed:
Update display_rgpd_consent.tpl

NicolasCador commented:
Hi @fox-john,

yes, you're right, huge job done !

I can re-propose my bug fix.

NicolasCador committed:
Delete front.js

NicolasCador committed:
Update ExportService.php

NicolasCador committed:
Update modules.tpl

NicolasCador commented:
Hi @hibatallahAouadni,

I updated my proposition following 2.0.1 release.

NicolasCador committed:
Update ExportService.php

NicolasCador commented:
Hi @fox-john,

ok, thank you very much, I updated.

matks reviewed (commented):
Hello @NicolasCador I try to review your Pull Request but it's very hard for me to understand the relationship between your code changes and the goal. Can you please guide me?

NicolasCador commented:
Hi @matks,
yes of course.
* I updated controllers/front/FrontAjaxGdpr.php to be in compliance with rules:
- ModuleFrontController instead of FrontController
- displayAjaxAddlog instead of if (Tools::getValue('action') !== 'AddLog') {...
* I updated controllers/front/gdpr.php and suppressed views/js/front.js because front.js is useless with just an ajax call with action=test never used anywhere and depending on a click on  something with id=exportPersonalData, implemented nowhere.
* I updated psgdpr.php and views/templates/hook/display_rgpd_consent.tpl to : 
- make easier the understanding of url
- ensure the ajax call will be done only on the right click with "parentForm.on('submit', function(event) {" instead of "$(document).on('submit', parentForm, function(event) {", because currently, if you click on "send" in contact form, it is also sending a second call for "newsletter inscription"
- make sure the ajax call will be done with success due to enough time: I added "async: false,". If not, the call is randomly OK, depending of the time spent of the call.
* Then I updated src/Service/ExportService.php and views/templates/front/pdf/sections/modules.tpl to allow developers to use their own Template for the pdf export, because it could be fun to be able to put data under a customized format especially when there are a lot of data to be exported. It's where the test fails... As we collect data through a json flow, to be retrocompatible I test if we have an array, it's as today without a template, if we have an object, I manage the template name. As the PHPStan test doesn't understand this subtlety, I think I will force the json collect as array, and then check if there is an index ['template'].

NicolasCador committed:
Update ExportService.php

NicolasCador committed:
Update ExportService.php

NicolasCador committed:
Update ExportService.php

NicolasCador commented:
Hi @kpodemski ,

ok, I added the 2 spaces required by the failed test.

Could you please relaunch a test?

Thanks

kpodemski commented:
@NicolasCador I just did, thanks Nicolas :)

NicolasCador committed:
Update ExportService.php

kpodemski commented:
@fox-john could you take a look and review the PR?

NicolasCador commented:
Hi @fox-john , @matks 

could you take a look and review the PR?

matks commented:
I will try to do it today or tomorrow ;)

matks reviewed (commented):
@NicolasCador OK I understood why I was confused ^^

I did not understand how the code changes you suggest would fix issue https://github.com/PrestaShop/PrestaShop/issues/31317 . But the issue https://github.com/PrestaShop/PrestaShop/issues/31317 is already solved: I just verified. I followed the issue steps and I am able to see the consent being registered.

So @NicolasCador your PR does not fix the problem (it is already fixed), it is now a refactoring. Is that right? You improve how the code is written but you do not modify the behavior.

NicolasCador commented:
Hi @matks,

there hasn't any change since my PR in the module code, so no, the https://github.com/PrestaShop/PrestaShop/issues/31317    is not already fixed:
While "$(document).on('submit', parentForm, function(event) {" is still for ajax call in views/templates/hook/display_rgpd_consent.tpl, if you have 2 modules on the same page (like ps_emailsubscription and productcomments) using psgdpr, the log will log 2 actions, while only one has been sent.
And while you don't have "async: false," in the ajax call, you are not sure the action will be logged (depending on the server speed).

As I explained in https://github.com/PrestaShop/psgdpr/pull/188#issuecomment-1545455010, the other codes are for cleaning, or make compliant with rules, and to allow developers to use their own Template for the pdf export, because it could be fun to be able to put data under a customized format especially when there are a lot of data to be exported.

We can have a small videoconference if necessary to discuss.

tashkasnamai commented:
![gdpr](https://github.com/PrestaShop/psgdpr/assets/150515341/1d3d9644-f3e0-4da6-b5b0-e0b2a0d501a7)
Can somebody fix this one, translations not working.

randshell commented:
Are there any updates on the progress of this issue?

tashkasnamai commented:
> Are there any updates on the progress of this issue?

yeah download from this page its fixed here

NicolasCador commented:
Hi all, where is 2.0.2 release?
